src/Stats/index.jsx

```diff
@@ -1,33 +1,63 @@
-import moment from 'moment';
-import utils from './utils';
 import styles from '../Common/themeStyle';
+import MainHeader from '../Common/MainHeader';
+import utils from './utils';
+import SubHeader from './DateSlider';
+import FilterBtn from './FilterBtn';
+import TransList from './TransList';
+import EmptyTrans from './EmptyTrans';
+
+const moment = require('moment');
 
 export default function Stats() {
   const { transactions } = useSelector((state) => state.transactions);
+  const [view, setCurrentView] = useState('month');
+  const [timePeriodOptions, setTimePeriod] = useState(utils.getDateSet(moment(), view));
+  const [isOverlayVisible, setOverlayVisibility] = useState(false);
+
+  const updateHeaderView = (type) => {
+    setCurrentView(type);
+    setTimePeriod(utils.getDateSet(moment(), type));
+    setOverlayVisibility(!isOverlayVisible);
+  };
+
+  const filterListView = (transactionRecords) => {
+    const current = timePeriodOptions[1];
+    return utils.filterData(transactionRecords, current, view);
+  };
+
   return (
-    <ScrollView>
-      <View style={styles.card}>
-        {utils.sortDataByDate(transactions).map((item, index) => (
-          <ListItem
-            key={`ListKey${index + 1}`}
-            title={item.label ? item.label : 'Undefined'}
-            subtitle={moment.unix(item.date).format('MM/DD/YYYY')}
-            rightTitle={
-              item.type === 'Income'
-                ? ` + $${item.amount}`
-                : ` - $${item.amount}`
-            }
-            leftIcon={{
-              title: 'Unknown',
-              name: 'attach-money',
-            }}
-          />
-        ))}
-      </View>
-    </ScrollView>
+    <View style={{ flex: 1, flexDirection: 'column' }}>
+      <Overlay height={200} isVisible={isOverlayVisible}>
+        <View>
+          <View style={{ flexDirection: 'row', justifyContent: 'flex-end' }}>
+            <Button icon={{ name: 'close' }} type="clear" onPress={() => setOverlayVisibility(!isOverlayVisible)} />
+          </View>
+          <View style={{ marginTop: 20 }}>
+            <ListItem title="Month" topDivider bottomDivider onPress={() => updateHeaderView('month')} />
+            <ListItem title="Year" bottomDivider onPress={() => updateHeaderView('year')} />
+          </View>
+        </View>
+      </Overlay>
+      <MainHeader title="Activity" onPressBtn={() => setOverlayVisibility(!isOverlayVisible)} />
+      <SubHeader
+        viewSet={timePeriodOptions}
+        onPressBtn={(value, type) => setTimePeriod(utils.getDateSet(value, type))}
+        viewType={view}
+      />
+      <ScrollView style={styles.deviceBody}>
+        <View style={{ marginBottom: 20 }}>
+          <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
+            <FilterBtn />
+          </View>
+          {(transactions.length !== 0)
+            ? <TransList transactions={filterListView(utils.groupData(transactions))} />
+            : <EmptyTrans />}
```

Comment:
EmptyTrans is a bit strange name, can we find a better? Like some placeholder, or empty history or whatever else?

Reply:
Update it as `EmptyHistory`